qbft/controller.go

```diff
@@ -227,14 +206,50 @@ func (c *Controller) Decode(data []byte) error {
 	}
 
 	config := c.GetConfig()
-	for _, i := range c.StoredInstances {
+	for _, i := range c.StoredInstances.All() {
 		if i != nil {
 			i.config = config
 		}
 	}
 	return nil
 }
 
+// UnmarshalJSON add a custom marshaling for Controller
+func (c *Controller) UnmarshalJSON(data []byte) error {
+	var objmap map[string]*json.RawMessage
+	if err := json.Unmarshal(data, &objmap); err != nil {
+		return err
+	}
+	fmt.Println(objmap)
+
+	var imc inMemContainer
```

Comment:
That shouldn't be here.. spec is not implementation specific

Reply:
but it is implemented an in memory before. the other option is to not use interface